extra_data/components.py

```diff
@@ -231,12 +253,15 @@ def get_dask_array(self, key, fill_value=None, astype=None):
 
         return self._concat(arrays, modnos, fill_value, astype)
 
-    def trains(self, require_all=True):
+    def trains(self, pulses=np.s_[:], require_all=True):
```

Comment:
This is in `MultimodDetectorBase` - you've added the pulses parameter here, making it equivalent to the method in `XtdfDetectorBase`, which you've then removed. However, if I'm reading the code in `MPxDetectorTrainIterator` correctly, passing `pulses=` will have no effect for JUNGFRAU - you'll get all pulses anyway.

I think we should put this and `XtdfDetectorBase.trains()` back as they are in master, at least for now. It's confusing to allow passing this parameter in for JUNGFRAU if it has no effect. If there are cases where it would be useful, we can come back and make it work at a later point.

Reply:
Sure, I'll put `XtdfDetectorBase.trains()` back.
I just thought to reduce repeated code, since `JUNGFRAU` class has its own `trains()` method without `pulses` parameter.

Reply:
Thanks!

My logic here is that `MultimodDetectorBase` doesn't 'know about' selecting pulses, so it shouldn't have parameters that do. If all the subclasses override it anyway, we could just have it `raise NotImplementedError` - but it's only a couple of lines of extra code, so it doesn't matter much.

Reply:
:+1: